data-prepper-plugins/mutate-string-processors/src/test/java/org/opensearch/dataprepper/plugins/processor/mutatestring/TruncateStringProcessorTests.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.mutatestring;
+
+import org.opensearch.dataprepper.expression.ExpressionEvaluator;
+import org.opensearch.dataprepper.metrics.PluginMetrics;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.model.record.Record;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Stream;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class TruncateStringProcessorTests {
+
+    @Mock
+    private PluginMetrics pluginMetrics;
+
+    @Mock
+    private TruncateStringProcessorConfig config;
+
+    @Mock
+    private ExpressionEvaluator expressionEvaluator;
+
+    private TruncateStringProcessor createObjectUnderTest() {
+        return new TruncateStringProcessor(pluginMetrics, config, expressionEvaluator);
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(TruncateStringArgumentsProvider.class)
+    void testTruncateStringProcessor(final String message, final int truncateLength, final String truncatedMessage) {
+
+        when(config.getIterativeConfig()).thenReturn(Collections.singletonList(createEntry("message", truncateLength, null)));
+
+        final TruncateStringProcessor truncateStringProcessor = createObjectUnderTest();
+        final Record<Event> record = createEvent(message);
+        final List<Record<Event>> truncatedRecords = (List<Record<Event>>) truncateStringProcessor.doExecute(Collections.singletonList(record));
+        assertThat(truncatedRecords.get(0).getData().get("message", Object.class), notNullValue());
+        assertThat(truncatedRecords.get(0).getData().get("message", Object.class), equalTo(truncatedMessage));
```

Comment:
I think this test is simple enough to use a mocked `Event`.

Then you just verify:

```
verify(truncatedRecords.get(0)).put("message", truncatedMessage);
```

Reply:
I think it is simple that we do not have to mock the event. I think it's better to not mock when the alternative is not hard.